### PR TITLE
wip

### PR DESCRIPTION
**Pull request by anhe-odoo**
## Description:

description of this task, what is implemented and why it is implemented that way.

Task: : [TASK_ID](https://www.odoo.com/web#id=TASK_ID&action=333&active_id=2328&model=project.task&view_type=form&cids=1&menu_id=4720)

## review checklist

- [ ] feature is organized in plugin, or UI components
- [ ] support of duplicate sheet (deep copy)
- [ ] in model/core: ranges are Range object, and can be adapted (adaptRanges)
- [ ] in model/UI: ranges are strings (to show the user)
- [ ] undo-able commands (uses this.history.update)
- [ ] multiuser-able commands (has inverse commands and transformations where needed)
- [ ] new/updated/removed commands are documented
- [ ] exportable in excel
- [ ] translations (\_lt("qmsdf %s", abc))
- [ ] unit tested
- [ ] clean commented code
- [ ] track breaking changes
- [ ] doc is rebuild (npm run doc)
- [ ] status is correct in Odoo

### CODE REVIEW

File: src/plugins/core/figures.ts [hunk 1]
```diff
@@ -59,6 +69,31 @@ export class FigurePlugin extends CorePlugin<FigureState> implements FigureState
     }
   }
 
+  copy(state: ClipboardState, isCutOperation: boolean) {
+    if (state instanceof ClipboardFigureState) {
+      this.copiedFigure = state.copiedFigure;
+      this.copiedFigureContent = state.copiedFigureContent;
+      this.operation = isCutOperation ? "CUT" : "COPY";
+    }
+  }
```
Comment (LucasLefevre) [inline src/plugins/core/figures.ts:78]: what I had in mind is that plugins would _return_ the clipped state and the clipboard itself would store it (without knowing exactly what's inside)
```suggestion
  copy(figureId, isCutOperation: boolean) {
    const copiedFigure = ...
    return {
       copiedFigure: ...
       copiedFigureContent: ...
    }
  }
```
Then, when calling the paste method, the clipboard would give the clipped content back
```ts
pasteFigure(sheetId: UID, position: { x: number; y: number }, clippedContent) {
   const copiedFigure = clippedContent.copiedFigure
   // do something
}
```

**Review by LucasLefevre — COMMENTED** [review]
I would go much further.

Basically everything in `ClipboardCellsState` should disappear and be moved to the respective plugins. `pasteCopiedTables` moves to the filter plugin,  `pasteCf` moves to the conditional format plugin, `pasteMergeIfExist` moves to the merge plugin, etc.

Same for the `ClipboardFigureState`, everything related to chart moves to the chart plugin, everything related to images moves to the image plugin.

Basically, delegate every business stuff back to plugin

**Comment by rrahir** [comment]
Done in [odoo/o-spreadsheet#2857](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-sandbox/workbench/workbench.html)
